pkg/helpers/newapp/app/sourcelookup.go

```diff
@@ -76,16 +76,11 @@ func IsRemoteRepository(s string) (bool, error) {
 		klog.V(5).Infof("%s is not a valid remote git clone spec", s)
 		return false, nil
 	}
-	gitRepo := git.NewRepository()
+	// don't need to check error since we already call s2igit.Parse
+	sri, _ := NewSourceRepository(s, newapp.StrategyUnspecified)
+	err = sri.DetectAuth()
 
-	// try up to 3 times to reach the remote git repo
-	for i := 0; i < 3; i++ {
-		_, _, err = gitRepo.ListRemote(url.StringNoFragment())
-		if err == nil {
-			break
-		}
-	}
-	if err != nil {
+	if err != nil || sri.requiresAuth {
```

Comment:
Should we have a different error message if the source repository requires authentication?
I think `could not list git remotes` would be confusing if the remote repository is otherwise valid.

Reply:
sure I'll separate out and have different logs (note level 5 verbosity) for each of these 2

Reply:
also note, testing proves that `DetachAuth` prints out explanatory logs prior to setting requiresAuth to true ... I'll factor that in to log here @adambkaplan